pandas/core/arrays/base.py

```diff
@@ -0,0 +1,201 @@
+"""An interface for extending pandas with custom arrays."""
+import abc
+
+import numpy as np
+
+from pandas.compat import add_metaclass
+
+
+_not_implemented_message = "{} does not implement {}."
+
+
+@add_metaclass(abc.ABCMeta)
+class ExtensionArray(object):
```

Comment:
Are there any expected requirements for the constructor `__init__`?

Reply:
We also expect that `ExtensionArray()`, with no arguments, works so that subclasses don't have to implement `construct_from_string`.

Rather than imposing that on subclasses, we could require some kind of `.empty` alternative constructor.